Userland/DynamicLoader/main.cpp

```diff
@@ -76,6 +80,30 @@ NAKED void _start(int, char**, char**)
 #endif
 }
 
+static ErrorOr<int> open_executable(char const* path)
+{
+    int rc = open(path, O_RDONLY | O_EXEC, 0666);
+    if (rc < 0)
+        return Error::from_errno(errno);
+    int checked_fd = rc;
+    ArmedScopeGuard close_on_failure([checked_fd] {
+        close(checked_fd);
+    });
+
+    struct stat executable_stat;
```

Comment:
let's zero initialize our stack variables before use, even if they're an out-param of a library function